nox/sessions.py

```diff
@@ -391,23 +416,44 @@ def run_always(
         if not args:
             raise ValueError("At least one argument required to run_always().")
 
-        return self._run(*args, env=env, **kwargs)
+        return self._run(
+            *args,
+            env=env,
+            include_invocation_env_vars=include_invocation_env_vars,
+            **kwargs,
+        )
 
     def _run(
-        self, *args: str, env: Mapping[str, str] | None = None, **kwargs: Any
+        self,
+        *args: str,
+        env: dict[str, str] | None = None,
+        include_invocation_env_vars: Iterable[str] | bool = True,
```

Comment:
I think this approach is worth discussing. In a way it is a subtle improvement over the suggestion in the bug thread (where passing an env var name with None as value meant "use the system". I also think it may more clear as it keeps this to just keys.

That said, I am not a huge fan of type of this var and that, in a way, it ends up overriding truthiness of the dict.

I think having the on/off makes a lot of sense, as this allows users of nox to control their sessions fully. If most folks are going to use this to bring 1-2 vars, it doesn't seem too overwhelming to need to do it via env:

```
result = session.run(
            ...
            env={"SYSTEM_ROOT": os.environ["SYSTEM_ROOT"]},
            ...
        )
```

That said, if it is more common for a user need many of them, this does cut down a bit on boilerplate. Though I can also imagine using a dict comp to do this in my code anyway:
```
env = {k: os.environ.get(k) for k in ["SYSTEM_ROOT", "LANGUAGE"]}
```

Reply:
I think the switch by itself is a good idea, potentially a rename to something a bit easier e.g. `include_outer`? But I agree on the type I think it should be just a bool on/off (with a default of `True` so the current default behaviour does not change).

I think 90%+ people who have a need to define env vars will have only a few. And if not the dictionary comp above works well, could even use a `.env` file or similar

Reply:
I'm also not a huge fan of maintaining a list of "special" env vars to always pass through, although I'm not entirely sure how else we'd do it